lib/array_intersection.rb

```diff
@@ -1,3 +1,19 @@
 def intersection(list1, list2)
+
+  i = 0
+
+  length_1 = list1.length
+  intersection = []
+
+  until i > length_1
+    list2.each do |num|
+      if list1[i] == num
+        intersection << list1[i]
+      end
```

Comment:
This is an O(n<sup>2</sup>) method.  You can do **much** better with a hash.  Think about it.